neqo-transport/src/connection/mod.rs

```diff
@@ -1134,20 +1139,21 @@ impl Connection {
             debug_assert!(self.crypto.states.rx_hp(self.version, cspace).is_some());
             for saved in self.saved_datagrams.take_saved() {
                 qtrace!([self], "input saved @{:?}: {:?}", saved.t, saved.d);
-                self.input(saved.d, saved.t, now);
+                self.input(&saved.d, saved.t, now);
             }
         }
     }
 
     /// In case a datagram arrives that we can only partially process, save any
     /// part that we don't have keys for.
-    fn save_datagram(&mut self, cspace: CryptoSpace, d: Datagram, remaining: usize, now: Instant) {
-        let d = if remaining < d.len() {
-            Datagram::new(d.source(), d.destination(), &d[d.len() - remaining..])
+    fn save_datagram(&mut self, cspace: CryptoSpace, d: &Datagram, remaining: usize, now: Instant) {
+        if remaining < d.len() {
+            let d = Datagram::new(d.source(), d.destination(), &d[d.len() - remaining..]);
+            self.saved_datagrams.save(cspace, d, now);
         } else {
-            d
+            // TODO: Sane?
+            self.saved_datagrams.save(cspace, d.clone(), now);
```

Comment:
By having `process_*` take a `&Datagram` instead of `Datagram`, one requires a `clone()` here.

See discussion in https://github.com/mozilla/neqo/issues/1538#issuecomment-1879731931.

Reply:
This is fine.  This is rare enough.